packages/connector-kakao/src/index.test.ts

```diff
@@ -0,0 +1,134 @@
+
+  describe('getAccessToken', () => {
+    afterEach(() => {
+      nock.cleanAll();
+      jest.clearAllMocks();
+    });
+
+    it('should get an accessToken by exchanging with code', async () => {
+      nock(accessTokenEndpoint).post('').reply(200, {
+        access_token: 'access_token',
+        scope: 'scope',
+        token_type: 'token_type',
+      });
+      const { accessToken } = await getAccessToken(mockedConfig, {
+        code: 'code',
+        redirectUri: 'dummyRedirectUri',
+      });
+      expect(accessToken).toEqual('access_token');
+    });
+
+    it('throws SocialAuthCodeInvalid error if accessToken not found in response', async () => {
+      nock(accessTokenEndpoint)
+        .post('')
+        .reply(200, { access_token: '', scope: 'scope', token_type: 'token_type' });
+      await expect(
+        getAccessToken(mockedConfig, { code: 'code', redirectUri: 'dummyRedirectUri' })
+      ).rejects.toMatchError(new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+    });
+  });
+
+  describe('getUserInfo', () => {
+    beforeEach(() => {
+      nock(accessTokenEndpoint).post('').reply(200, {
+        access_token: 'access_token',
+        scope: 'scope',
+        token_type: 'token_type',
+      });
+    });
+
+    afterEach(() => {
+      nock.cleanAll();
+      jest.clearAllMocks();
+    });
+
+    it('should get valid SocialUserInfo', async () => {
+      nock(userInfoEndpoint).post('').reply(200, {
+        sub: '1234567890',
+        name: 'monalisa octocat',
+        given_name: 'monalisa',
+        family_name: 'octocat',
+        picture: 'https://github.com/images/error/octocat_happy.gif',
+        email: 'octocat@google.com',
+        email_verified: true,
+        locale: 'en',
+      });
+      const connector = await createConnector({ getConfig });
+      const socialUserInfo = await connector.getUserInfo({
+        code: 'code',
+        redirectUri: 'redirectUri',
+      });
+      expect(socialUserInfo).toMatchObject({
+        id: '1234567890',
+        avatar: 'https://github.com/images/error/octocat_happy.gif',
+        name: 'monalisa octocat',
+        email: 'octocat@google.com',
+      });
+    });
+
+    it('throws SocialAccessTokenInvalid error if remote response code is 401', async () => {
+      nock(userInfoEndpoint).post('').reply(401);
+      const connector = await createConnector({ getConfig });
+      await expect(connector.getUserInfo({ code: 'code', redirectUri: '' })).rejects.toMatchError(
+        new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid)
+      );
+    });
+
+    it('throws General error', async () => {
+      nock(userInfoEndpoint).post('').reply(200, {
+        sub: '1234567890',
+        name: 'monalisa octocat',
+        given_name: 'monalisa',
+        family_name: 'octocat',
+        picture: 'https://github.com/images/error/octocat_happy.gif',
+        email: 'octocat@google.com',
+        email_verified: true,
+        locale: 'en',
+      });
```

Comment:
nit: As @simeng-li  said, the mocked response does not fit the Kakao endpoint response.